File: back/app/entity/Task.go
package entity

import (
	"Todolist/app/orm"
	"database/sql"
)

type Task struct {
	Id        int           `json:"id"         params:"type: int, primary_key: true, auto_increment: true"`
	Amount    sql.NullInt16 `json:"amount"     params:"type: int, nullable: true"`
	Completed int           `json:"completed"  params:"type: int, nullable: false"`
	Priority  int           `json:"priority"   params:"type: int, nullable: false"`
	TaskGroup int           `json:"task_group" params:"type: int, nullable: false, foreign_key: task_group"`
	Name      string        `json:"name"       params:"type: varchar, length: 255, nullable: false"`
	Type      string        `json:"type"       params:"type: varchar, length: 255, nullable: false"`
	CustomId  string        `json:"custom_id"  params:"type: varchar, length: 255, nullable: false"`
}

func (t *Task) GetTaskGroup() *TaskGroup {
	return orm.FindById(TaskGroup{}, t.TaskGroup).(*TaskGroup)
}

File: front/app/utils/Task.ts
import { PriorityObjectType, PriorityObjects } from '../types/PriorityType';
import { JSONTask, Task } from '../types/TaskType';

export function JsonToTask(json: JSONTask): Task {
  const task = {
    name: json.name,
    task_group: json.task_group,
    priority: Object.values(PriorityObjects).find((priority) => priority.id === json.priority) as PriorityObjectType,
  };

  if (json.type === 'quantity') {
    return {
      ...task,
      type: json.type,
      amount: json.amount,
      completed: Number(json.completed),
    };
  } else {
    return {
      ...task,
      type: json.type,
      completed: json.completed === 1 ? true : false,
    };
  }
}
